app/web/features/userQueries/useLiteUsers.test.tsx

```diff
@@ -0,0 +1,283 @@
+        }
+      );
+
+      await waitForNextUpdate();
+      expect(mockGetLiteUsers).toHaveBeenCalledTimes(1);
+
+      expect(result.current).toEqual(
+        expect.objectContaining({
+          data: new Map(mockLiteUsers.map((user) => [user.userId, user])),
+          error: null,
+          isError: false,
+          isFetching: false,
+          isLoading: false,
+          isRefetching: false,
+        })
+      );
+    });
+
+    it("Should filter out undefined ids", async () => {
+      const idsWithUndefined = [1, undefined, 3];
+
+      const { result, waitFor } = renderHook(
+        () => useLiteUsers(idsWithUndefined),
+        { wrapper }
+      );
+
+      await waitFor(() => expect(result.current.isLoading).toBe(false));
+
+      expect(mockGetLiteUsers).toHaveBeenCalledWith([1, 3]);
+    });
+
+    it("Should return an error if the query fails", async () => {
+      mockConsoleError();
+      const error = new RpcError(500, "Some error", {});
+      mockGetLiteUsers.mockRejectedValue(error);
+
+      const { result, waitForNextUpdate } = renderHook(
+        () => useLiteUsers([1, 2, 3]),
+        { wrapper }
+      );
+
+      await waitForNextUpdate();
+
+      expect(result.current.isLoading).toBe(false);
+      expect(result.current.error).toBe(error);
+      expect(result.current.data).toBeUndefined();
+    });
+
+    it("Should not run the query when ids is an empty array", async () => {
+      const ids: number[] = [];
+
+      renderHook(() => useLiteUsers(ids), { wrapper });
+
+      // Ensure the query function was NOT called
+      expect(mockGetLiteUsers).not.toHaveBeenCalled();
+    });
+
+    it("Should not run the query function when ids is undefined", () => {
+      const ids: (number | undefined)[] = [undefined];
+
+      renderHook(() => useLiteUsers(ids), { wrapper });
+
+      expect(mockGetLiteUsers).not.toHaveBeenCalled();
+    });
+  });
+
+  describe("useLiteUsers - cached data", () => {
+    const mockGetLiteUsers = service.user.getLiteUsers as jest.Mock;
+
+    const ids = [1, 2, 3, 4];
+
+    const sharedClient = new QueryClient({
+      defaultOptions: { queries: { retry: false } },
+    });
+
+    const sharedClientWrapper = ({
+      children,
+    }: {
+      children: React.ReactNode;
+    }) => (
+      <QueryClientProvider client={sharedClient}>
+        {children}
+      </QueryClientProvider>
+    );
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+
+      sharedClient.clear();
+      // Set the cached data for the specific liteUserIds
+      sharedClient.setQueryData(liteUsersKey(ids), getLiteUsers(ids));
+    });
+
+    it("Uses cached data instead of refetching", async () => {
+      const { result } = renderHook(() => useLiteUsers(ids), {
+        wrapper: sharedClientWrapper,
+      });
+
+      expect(mockGetLiteUsers).not.toHaveBeenCalled();
+
+      expect(result.current.isFetching).toBe(false);
+      expect(result.current.isLoading).toBe(false);
+    });
```

Comment:
I think this should also assert `result.current.data` exist to prove it has data and is reading from the cache

Reply:
I ended up going about this another way, by calling the hook twice and checking that the second time used cache data (rather than setting it manually in the queryClient).